src/components/Main/Main.tsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  visibleTodos: Todo[],
+  onRemoveTodo: (todo: Todo, index: number) => void
+  onEditTodo: (
+    event: React.ChangeEvent<HTMLInputElement>,
+    todoId: number
+  ) => void,
+  hasEditTodo: boolean,
+  setHasEditTodo: React.Dispatch<React.SetStateAction<boolean>>,
+  onUpdateTodo: (todo: Todo) => Promise<void>,
+  onChangeStatusTodo: (todoId: number) => void,
+  todoForUpdate: Todo | null,
+  setTodoForUpdate: React.Dispatch<React.SetStateAction<Todo | null>>,
+  idTodoForCheange: number[],
+};
+
+export const Main: React.FC<Props> = ({
+  visibleTodos,
+  onRemoveTodo,
+  onEditTodo,
+  hasEditTodo,
+  setHasEditTodo,
+  onUpdateTodo,
+  onChangeStatusTodo,
+  todoForUpdate,
+  setTodoForUpdate,
+  idTodoForCheange,
+}) => {
+  return (
+    <section className="todoapp__main">
+      {visibleTodos.map((todo, index) => {
+        if (idTodoForCheange.includes(todo.id)) {
+          return (
+            <div className="todo" key={`todo-${todo.id}`}>
+              <label className="todo__status-label">
+                <input type="checkbox" className="todo__status" />
+              </label>
+              <span
+                className="todo__title"
+              >
+                {todo.title}
+              </span>
+              <button
+                type="button"
+                className="todo__remove"
+              >
+                ×
+              </button>
+
+              <div className="modal overlay is-active">
+                <div className="modal-background has-background-white-ter" />
+                <div className="loader" />
+              </div>
+            </div>
+          );
+        }
+
+        return (
+          <div
+            className={cn(
+              'todo',
+              { completed: todo.completed },
+            )}
+            key={todo.id}
+          >
+            <label className="todo__status-label">
+              <input
+                type="checkbox"
+                className="todo__status"
+                onClick={() => (onChangeStatusTodo(todo.id))}
+              />
+            </label>
+
+            {hasEditTodo && todoForUpdate?.id === todo.id
+              ? (
+                <>
+                  <form onSubmit={(
+                    event: {
+                      preventDefault: () => void;
+                      stopPropagation: () => void;
```

Comment:
why you don't use `FormEvent` here
```suggestion
                  <form onSubmit={(
                    event: {
                      preventDefault: () => void;
                      stopPropagation: () => void;
```